pandas/plotting/_matplotlib/core.py

```diff
@@ -489,6 +489,18 @@ def _post_plot_logic(self, ax, data):
         """Post process for each axes. Overridden in child classes"""
         pass
 
+    def _has_iterable_only_float_int_or_datetime(self, list_iterable: list) -> bool:
```

Comment:
_has_valid_dtype

remove the double-doc-string

fully type list

Reply:
btw, small idea on the name, could you maybe use `_has_valid_lim_dtype`? this is only targeting on dtypes of `xlim` or `ylim`, right?  @regmibijay